source/material_model/rheology/compositional_viscosity_prefactors.cc

```diff
@@ -0,0 +1,142 @@
+#include <aspect/global.h>
+#include <aspect/geometry_model/interface.h>
+
+#include <deal.II/base/signaling_nan.h>
+#include <deal.II/base/parameter_handler.h>
+#include <aspect/simulator_signals.h>
+#include <aspect/parameters.h>
+
+
+// TODO: Add constant_viscosity_prefactors to this file at a point when ASPECT is ready to be broken
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    namespace Rheology
+    {
+      template <int dim>
+      CompositionalViscosityPrefactors<dim>::CompositionalViscosityPrefactors ()
+        = default;
+
+      template <int dim>
+      double
+      CompositionalViscosityPrefactors<dim>::compute_viscosity (const MaterialModel::MaterialModelInputs<dim> &in,
+                                                                const double base_viscosity,
+                                                                const unsigned int composition_index,
+                                                                const unsigned int q) const
+      {
+        double prefactor;
+        switch (viscosity_prefactor_scheme)
+          {
+            case none:
+            {
+              prefactor = 1;
+              break;
+            }
+            case water_fugacity:
+            {
+              const unsigned int bound_fluid_idx = this->introspection().compositional_index_for_name("bound_fluid");
+              const double wt_fraction_h2o = in.composition[q][bound_fluid_idx];
+              const double wt_ol = 1 - wt_fraction_h2o; // We need to get wt_h2o from bound water
+              const double COH = (wt_fraction_h2o/M_h2o) / (wt_ol/M_ol) * 1e6;
+              const double point_water_fugacity = COH / A_h2o * \
+                                                  std::exp((E_h2o + in.pressure[q]*V_h2o)/(constants::gas_constant * in.temperature[q]));
+              prefactor = std::pow(point_water_fugacity, -water_fugacity_exponents[composition_index]);
+            }
+          }
+
+        return prefactor * base_viscosity;
+      }
+
+
+      template <int dim>
+      void
+      CompositionalViscosityPrefactors<dim>::declare_parameters (ParameterHandler &prm)
+      {
+        prm.declare_entry ("Exponents for water fugacity", "0.0",
+                           Patterns::List(Patterns::Double(0)),
+                           "List of viscosity prefactors (i.e., multiplicative factors) "
+                           "for background material and compositional fields, for a total of N+1 "
+                           "where N is the number of all compositional fields or only those "
+                           "corresponding to chemical compositions. Units: none.");
+
+        prm.declare_entry ("Viscosity prefactor scheme", "none",
+                           Patterns::Selection("none|water fugacity|"),
+                           "Select what type of multiplicative prefactor you want to apply to the viscosity. "
+                           "This is applied before yielding. Units: none.");
```

Comment:
```suggestion
                           "Units: none.");
```
I think the yielding part should be removed (again, I realize this is legacy documentation).